BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -100,8 +100,14 @@ private XDocument LoadPostXml(string filePath)
         {
             string text = _fileSystem.ReadFileText(filePath);
             StringReader reader = new StringReader(text);
-
-            return XDocument.Load(reader);
+            try
+            {
+                return XDocument.Load(reader);
+            }
+            catch
+            {
+                throw new NullReferenceException();

```

Comment:
NullReferenceException is not the right type to throw here.  XDocument.Load is throwing a better one.

In this case, who is catching the exception you throw?  I think you just want to catch what XDocument.Load throws and move on.  (Ideally we'd be logging something, but we can leave that for another time).